tests/jax/components/building/adders_test.py

```diff
@@ -15,135 +15,211 @@
+
+    parallel_sequence_adder.on_building_init_start(builder=mock_builder)
+    assert (
+        mock_builder.store.sequence_length
+        == parallel_sequence_adder.config.sequence_length
     )
-    test_system_parallel_sequence_adder._builder.data_server()
-    test_system_parallel_sequence_adder._builder.store.system_executor = None
-    test_system_parallel_sequence_adder._builder.executor(
-        executor_id="executor", data_server_client=None, parameter_server_client=None
+    parallel_sequence_adder.on_building_executor_adder(builder=mock_builder)
+    assert type(mock_builder.store.adder) == reverb_adders.ParallelSequenceAdder
+    assert parallel_sequence_adder.name() == "executor_adder"
+
+
+def test_sequence_adders_signature(
+    mock_builder: Builder,
+    parallel_sequence_adder_signature: ParallelSequenceAdderSignature,
+) -> None:
+    """Test sequence adder signature callback.
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_sequence_adder_signature: Fixture ParallelSequenceAdderSignature.
+
+    Returns:
+        None
+    """
+    parallel_sequence_adder_signature.on_building_data_server_adder_signature(
+        builder=mock_builder
+    )
+    assert mock_builder.store.adder_signature_fn
+    assert parallel_sequence_adder_signature.name() == "data_server_adder_signature"
+    assert issubclass(
+        parallel_sequence_adder_signature.config_class(),  # type: ignore
+        AdderSignatureConfig,
+    )
+
+
+def test_transition_adders(
+    mock_builder: Builder,
+    parallel_transition_adder: ParallelTransitionAdder,
+) -> None:
+    """Test transition adder callback
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_transition_adder: Fixture ParallelTransitionAdder.
+
+    Returns:
+        None
+    """
+    parallel_transition_adder.on_building_executor_adder(builder=mock_builder)
+    assert type(mock_builder.store.adder) == reverb_adders.ParallelNStepTransitionAdder
+    assert parallel_transition_adder.name() == "executor_adder"
+
+
+def test_transition_adders_signature(
+    mock_builder: Builder,
+    parallel_transition_adder_signature: ParallelTransitionAdderSignature,
+) -> None:
+    """Test transition adder signature callback
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_transition_adder_signature: Fixture ParallelTransitionAdderSignature.
+
+    Returns:
+        None
+    """
+    parallel_transition_adder_signature.on_building_data_server_adder_signature(
+        builder=mock_builder
     )
-    test_system_parallel_sequence_adder._builder.store.adder.add_first(env_restart)
-    test_system_parallel_sequence_adder._builder.store.server.stop()
+    assert mock_builder.store.adder_signature_fn
+    assert parallel_transition_adder_signature.name() == "data_server_adder_signature"
+    assert issubclass(
+        parallel_transition_adder_signature.config_class(),  # type: ignore
+        AdderSignatureConfig,
+    )
```

Comment:
I get that we create the config and component and assign them in the same class so perhaps this is overkill but isn't it worth checking in case somebody assigned the wrong type config class to the component?

Reply:
I have thought about this and the fact that we create the config class and assign it means that we are free to set it to what we want hence removed the asserts